pyscf/dft/uks.py

```diff
@@ -34,16 +35,32 @@ def get_veff(ks, mol=None, dm=None, dm_last=0, vhf_last=0, hermi=1):
         if ks.small_rho_cutoff > 1e-20 and ground_state:
             ks.grids = rks.prune_small_rho_grids_(ks, mol, dm[0]+dm[1], ks.grids)
         t0 = logger.timer(ks, 'setting up grids', *t0)
+    if ks.nlc!='':
+        if ks.nlcgrids.coords is None:
+            ks.nlcgrids.build(with_non0tab=True)
+            if ks.small_rho_cutoff > 1e-20 and ground_state:
+                ks.nlcgrids = rks.prune_small_rho_grids_(ks, mol, dm[0]+dm[1], ks.nlcgrids)
+            t0 = logger.timer(ks, 'setting up nlc grids', *t0)
 
     if hermi == 2:  # because rho = 0
         n, exc, vxc = (0,0), 0, 0
     else:
         n, exc, vxc = ks._numint.nr_uks(mol, ks.grids, ks.xc, dm)
+        if ks.nlc!='':
+            _, enlc, vnlc = ks._numint.nr_rks(mol, ks.nlcgrids, ks.xc+'__'+ks.nlc, dm[0]+dm[1])
+            exc += enlc
+            vxc += vnlc
         logger.debug(ks, 'nelec by numeric integration = %s', n)
         t0 = logger.timer(ks, 'vxc', *t0)
 
     hyb = ks._numint.hybrid_coeff(ks.xc, spin=mol.spin)
-    if abs(hyb) < 1e-10:
+
+    #enabling range-separated hybrids
+    omega, alpha, beta = ks._numint.rsh_coeff(ks.xc)
+    if abs(omega) > 1e-10:
+        hyb = alpha + beta
```

Comment:
What's the point of this? In libxc notation, alpha is full Hartree-Fock exchange, and beta is short-range exchange.

Reply:
No it is not. CAM-B3LYP, for example, has 19%/65% SR/LR, and returns this:
alpha = 0.65;
beta  =-0.46;
Clearly, SR=alpha+beta, LR=alpha.

Reply:
Well, you're not wrong. As I said, alpha is full Hartree-Fock exchange, and beta is short-range only. Thus, in the long range you get just alpha. And in the short range, you get alpha+beta. I'm just saying that ```hyb``` is misleading, since that's the (non-range-separated) hybrid coefficient which is just alpha.